sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2871,7 +2872,7 @@ case class ToBinary(
             errorSubClass = "NON_FOLDABLE_INPUT",
             messageParameters = Map(
               "inputName" -> toSQLId("fmt"),
-              "inputType" -> toSQLType(StringType),
+              "inputType" -> toSQLType(StringTypeAnyCollation),
```

Comment:
Why not f.dataType here as you did for INVALID_ARG_VALUE

Reply:
Template for `INVALID_ARG_VALUE`: https://github.com/apache/spark/blob/dd4c7fc6ae91ec1f2a348cf4633bff10717cde22/common/utils/src/main/resources/error/error-classes.json#L746
Template for `NON_FOLDABLE_INPUT`: https://github.com/apache/spark/blob/dd4c7fc6ae91ec1f2a348cf4633bff10717cde22/common/utils/src/main/resources/error/error-classes.json#L806
In case of `INVALID_ARG_VALUE`, the `inputValue` should be what the input data type is, in this case `f.dataType`, while in case of `NON_FOLDABLE_INPUT`, `inputType` is what we're expecting and that's `StringTypeAnyCollation`.